Zotero/Scenes/Detail/PDF/Views/PDFSidebarViewController.swift

```diff
@@ -56,91 +64,132 @@ class PDFSidebarViewController: UIViewController {
-            self.view.trailingAnchor.constraint(equalTo: container.trailingAnchor),
-            container.topAnchor.constraint(equalTo: self.annotationsController.view.topAnchor),
-            container.bottomAnchor.constraint(equalTo: self.annotationsController.view.bottomAnchor),
-            container.leadingAnchor.constraint(equalTo: self.annotationsController.view.leadingAnchor),
-            container.trailingAnchor.constraint(equalTo: self.annotationsController.view.trailingAnchor),
-            container.topAnchor.constraint(equalTo: self.outlineController.view.topAnchor),
-            container.bottomAnchor.constraint(equalTo: self.outlineController.view.bottomAnchor),
-            container.leadingAnchor.constraint(equalTo: self.outlineController.view.leadingAnchor),
-            container.trailingAnchor.constraint(equalTo: self.outlineController.view.trailingAnchor)
+            picker.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 15),
+            view.trailingAnchor.constraint(equalTo: picker.trailingAnchor, constant: 15),
+            view.bottomAnchor.constraint(equalTo: container.bottomAnchor),
+            view.leadingAnchor.constraint(equalTo: container.leadingAnchor),
+            view.trailingAnchor.constraint(equalTo: container.trailingAnchor),
+            container.topAnchor.constraint(equalTo: thumbnailsController.view.topAnchor),
+            container.bottomAnchor.constraint(equalTo: thumbnailsController.view.bottomAnchor),
+            container.leadingAnchor.constraint(equalTo: thumbnailsController.view.leadingAnchor),
+            container.trailingAnchor.constraint(equalTo: thumbnailsController.view.trailingAnchor),
+            container.topAnchor.constraint(equalTo: annotationsController.view.topAnchor),
+            container.bottomAnchor.constraint(equalTo: annotationsController.view.bottomAnchor),
+            container.leadingAnchor.constraint(equalTo: annotationsController.view.leadingAnchor),
+            container.trailingAnchor.constraint(equalTo: annotationsController.view.trailingAnchor),
+            container.topAnchor.constraint(equalTo: outlineController.view.topAnchor),
+            container.bottomAnchor.constraint(equalTo: outlineController.view.bottomAnchor),
+            container.leadingAnchor.constraint(equalTo: outlineController.view.leadingAnchor),
+            container.trailingAnchor.constraint(equalTo: outlineController.view.trailingAnchor)
         ])
     }
 
     private func setupControllers() {
-        let annotationsController = AnnotationsViewController(viewModel: self.viewModel)
-        annotationsController.parentDelegate = self.parentDelegate
-        annotationsController.coordinatorDelegate = self.coordinatorDelegate
-        annotationsController.boundingBoxConverter = self.boundingBoxConverter
-        self.addChild(annotationsController)
+        let annotationsController = AnnotationsViewController(viewModel: viewModel)
```

Comment:
Since we create several controllers when the document appears, that may not even be used at all, maybe we can convert them to lazy load, in a future update. Or maybe just pre-load the first, or some.

Reply:
I considered this to be pretty much the same as `UITabbarController` which allocates all controllers ahead too. But we can definitely improve this by lazy loading.